src/Rings/msolve/msolve.jl

```diff
@@ -0,0 +1,213 @@
+```
+"""
+function msolve(
+        I::MPolyIdeal;                        # input generators
+        initial_hts::Int=17,                  # hash table size, default 2^17
+        nr_thrds::Int=1,                      # number of threads
+        max_nr_pairs::Int=0,                  # number of pairs maximally chosen
+                                              # in symbolic preprocessing
+        la_option::Int=2,                     # linear algebra option
+        info_level::Int=0,                    # info level for print outs
+        precision::Int=32,                    # precision of the solution set
+        get_param::Bool=false                 # return rational parametrization of
+                                              # solution set
+        )
+    singular_assure(I)
+    SI    = I.gens.S
+    R     = I.gens.Sx
+    # skip zero generators in ideal
+    ptr = Singular.libSingular.id_Copy(SI.ptr, R.ptr)
+    J   = Singular.Ideal(R, ptr)
+    Singular.libSingular.idSkipZeroes(J.ptr)
+    # get number of variables
+    nr_vars = Singular.nvars(R)
+    nr_gens = Singular.ngens(J)
+    vars    = Singular.gens(R)
+
+    variable_names  = Array{String, 1}(undef, nr_vars)
+    for i in 1:nr_vars
+        variable_names[i] = string(Singular.gens(R)[i])
+    end
+
+    field_char  = Singular.characteristic(R)
+
+    #= add new variables and linear forms, =#
+    genericity_handling = 2
+
+    reset_ht  = 0
+    print_gb  = 0
+
+    #= monomial order defaults to zero =#
+    mon_order = 0
+
+    # convert Singular ideal to flattened arrays of ints
+    if 0 == field_char
+      lens, cfs, exps   = convert_qq_singular_ideal_to_array(J, nr_vars, nr_gens)
+    # elseif isprime(field_char)
+    #   lens, cfs, exps   = convert_ff_singular_ideal_to_array(J, nr_vars, nr_gens)
+    else
+        # error("At the moment GroebnerBasis only supports finite fields and the rationals.")
+        @error "At the moment msolve only supports the rationala as ground field."
+    end
+    # lib = Libdl.dlopen("/home/ederc/repos/master-msolve/src/msolve/.libs/libmsolve-0.1.2.so")
+    lib = Libdl.dlopen(libmsolve)
+    sym = Libdl.dlsym(lib, :msolve_julia)
+
+    res_ld    = ccall(:malloc, Ptr{Cint}, (Csize_t, ), sizeof(Cint))
+    res_dim   = ccall(:malloc, Ptr{Cint}, (Csize_t, ), sizeof(Cint))
+    res_dquot = ccall(:malloc, Ptr{Cint}, (Csize_t, ), sizeof(Cint))
+    res_len   = ccall(:malloc, Ptr{Ptr{Cint}}, (Csize_t, ), sizeof(Ptr{Cint}))
+    res_cf    = ccall(:malloc, Ptr{Ptr{Cvoid}}, (Csize_t, ), sizeof(Ptr{Cvoid}))
+    nb_sols   = ccall(:malloc, Ptr{Cint}, (Csize_t, ), sizeof(Cint))
+    sols_num  = ccall(:malloc, Ptr{Ptr{Cvoid}}, (Csize_t, ), sizeof(Ptr{Cvoid}))
+    sols_den  = ccall(:malloc, Ptr{Ptr{Cint}}, (Csize_t, ), sizeof(Ptr{Cint}))
+    ccall(sym, Cvoid,
```

Comment:
```suggestion
    ccall((:msolve_julia, libmsolve), Cvoid,
```

Reply:
Thanks, I have taken your changes. Haven't known about this, sorry.